### PR TITLE
feat(utxo-bin): use prevOutputs, spend status

### PR DESCRIPTION
**Pull request by OttoAllmendinger**
* allow lookup via txid
* allow transaction status lookup (found/confirmed)
* allow transaction input lookup (signature verification, fee info)
* allow transaction output spend status lookup

Issue: BG-41897

### CODE REVIEW

File: modules/statics/src/map.ts [hunk 1]
```diff
@@ -59,4 +59,8 @@ export class CoinMap {
   public forEach(callback: (coin: Readonly<BaseCoin>, coinName: string) => void): void {
     this._map.forEach(callback);
   }
+
+  public [Symbol.iterator](): IterableIterator<[string, Readonly<BaseCoin>]> {
```
Comment (sangaman) [inline modules/statics/src/map.ts:63]: What is this needed/used for?

File: modules/utxo-bin/src/commands.ts [hunk 2]
```diff
@@ -7,25 +7,44 @@ import * as utxolib from '@bitgo/utxo-lib';
...
 
 export const cmdParse = {
-  command: 'parse <path>',
+  command: 'parse [path]',
   describe: 'display transaction components in human-readable form',
 
   builder(b: yargs.Argv<unknown>): yargs.Argv<Args> {
     return b
-      .option('path', { type: 'string', default: '-' })
+      .option('path', { type: 'string', nargs: 1, default: '' })
```
Comment (sangaman) [inline modules/utxo-bin/src/commands.ts:41]: Why is `nargs` and `default` set here? Would this option just default to `undefined` without it?

**Review by sangaman — APPROVED** [review]
Nice work, this is a great tool.

File: modules/statics/src/map.ts (hunk 1 above)
Comment (OttoAllmendinger) [reply modules/statics/src/map.ts:63]: https://github.com/BitGo/BitGoJS/pull/1923/commits/a4a2f4b830084a136840abbaf3b30fe5852a60e1

File: modules/utxo-bin/src/commands.ts (hunk 2 above)
Comment (OttoAllmendinger) [reply modules/utxo-bin/src/commands.ts:41]: This is to make `yargs parse -` actually parse stdin.  Without this the, `-` argument is ignored: https://github.com/yargs/yargs/issues/1312#issuecomment-940633564

**Pull request merged**